diff -ruN a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2020-03-21 06:37:32.711158734 +0100
+++ b/kernel/conftest.sh	2020-03-21 07:42:40.184633621 +0100
@@ -465,16 +465,12 @@
             # It does not exist on all architectures.
             #
             CODE="
-            #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
-            #include <asm/set_memory.h>
-            #else
             #include <asm/cacheflush.h>
-            #endif
             void conftest_set_memory_array_uc(void) {
                 set_memory_array_uc();
             }"
 
-            compile_check_conftest "$CODE" "NV_SET_MEMORY_ARRAY_UC_PRESENT" "" "functions"
+            compile_check_conftest "$CODE" "NV_SET_MEMORY_ARRAY_UC_NOT_PRESENT" "" "functions"
         ;;
 
         sysfs_slab_unlink)
diff -ruN a/kernel/nvidia/nv-vm.c b/kernel/nvidia/nv-vm.c
--- a/kernel/nvidia/nv-vm.c	2020-03-21 06:37:32.661161700 +0100
+++ b/kernel/nvidia/nv-vm.c	2020-03-21 07:34:41.795459362 +0100
@@ -70,10 +70,11 @@
             break;
 #endif
         default:
-            nv_printf(NV_DBG_ERRORS,
+/*            nv_printf(NV_DBG_ERRORS,
                 "NVRM: %s(): type %d unimplemented\n",
                 __FUNCTION__, type);
-            break;
+*/
+              break;
     }
 }
 
