diff -ruN a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
--- a/kernel/nvidia-drm/nvidia-drm-drv.c	2020-07-12 09:34:11.888138363 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c	2020-07-12 10:11:45.086159060 +0200
@@ -719,7 +719,7 @@
 #endif
                                DRIVER_GEM  | DRIVER_RENDER,
 
-    .gem_free_object        = nv_drm_gem_free,
+    .gem_free_object_unlocked = nv_drm_gem_free,
 
     .ioctls                 = nv_drm_ioctls,
     .num_ioctls             = ARRAY_SIZE(nv_drm_ioctls),
diff -ruN a/kernel/nvidia-drm/nvidia-drm-gem.h b/kernel/nvidia-drm/nvidia-drm-gem.h
--- a/kernel/nvidia-drm/nvidia-drm-gem.h	2020-07-12 09:34:11.888138363 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-gem.h	2020-07-12 09:50:36.410155340 +0200
@@ -96,7 +96,7 @@
 nv_drm_gem_object_unreference_unlocked(struct nv_drm_gem_object *nv_gem)
 {
 #if defined(NV_DRM_GEM_OBJECT_GET_PRESENT)
-    drm_gem_object_put_unlocked(&nv_gem->base);
+    drm_gem_object_put(&nv_gem->base);
 #else
     drm_gem_object_unreference_unlocked(&nv_gem->base);
 #endif
