diff -ruN a/kernel/common/inc/nv-mm.h b/kernel/common/inc/nv-mm.h
--- a/kernel/common/inc/nv-mm.h	2020-07-12 09:34:11.898127622 +0200
+++ b/kernel/common/inc/nv-mm.h	2020-07-12 09:35:29.156463757 +0200
@@ -141,13 +141,13 @@
 
         #if defined(NV_GET_USER_PAGES_REMOTE_HAS_LOCKED_ARG)
 
-               return get_user_pages_remote(tsk, mm, start, nr_pages, flags,
-                                            pages, vmas, NULL);
+               return get_user_pages_remote(mm, start, nr_pages, flags,
+                                            pages, vmas, NULL, NULL);
 
         #else
 
-               return get_user_pages_remote(tsk, mm, start, nr_pages, flags,
-                                            pages, vmas);
+               return get_user_pages_remote(mm, start, nr_pages, flags,
+                                            pages, vmas, NULL);
 
         #endif
 
@@ -176,7 +176,7 @@
             if (force)
                 flags |= FOLL_FORCE;
 
-            return get_user_pages(tsk, mm, start, nr_pages, flags, pages, vmas);
+            return get_user_pages(mm, start, nr_pages, flags, pages, vmas, NULL);
         }
     #endif
 #endif
