diff -ruN a/kernel/nvidia-drm/nvidia-drm-utils.c b/kernel/nvidia-drm/nvidia-drm-utils.c
--- a/kernel/nvidia-drm/nvidia-drm-utils.c	2020-07-12 16:40:48.311623803 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-utils.c	2020-07-12 17:08:58.987889082 +0200
@@ -103,7 +103,7 @@
 nvkms_display_mode_to_drm_mode(const struct NvKmsKapiDisplayMode *displayMode,
                                struct drm_display_mode *mode)
 {
-    mode->vrefresh    = (displayMode->timings.refreshRate + 500) / 1000; /* In Hz */
+/*    mode->vrefresh    = (displayMode->timings.refreshRate + 500) / 1000; /* In Hz */
 
     mode->clock       = (displayMode->timings.pixelClockHz + 500) / 1000; /* In Hz */
 
@@ -159,7 +159,7 @@
 void drm_mode_to_nvkms_display_mode(const struct drm_display_mode *src,
                                     struct NvKmsKapiDisplayMode *dst)
 {
-    dst->timings.refreshRate  = src->vrefresh * 1000;
+/*    dst->timings.refreshRate  = src->vrefresh * 1000;*/
 
     dst->timings.pixelClockHz = src->clock * 1000; /* In Hz */
 
