diff -ruN a/kernel/nvidia/os-interface.c b/kernel/nvidia/os-interface.c
--- a/kernel/nvidia/os-interface.c	2020-01-23 08:18:47.818433061 +0100
+++ b/kernel/nvidia/os-interface.c	2020-01-23 08:19:22.247595009 +0100
@@ -477,9 +477,9 @@
 
 void NV_API_CALL os_get_current_tick(NvU64 *nseconds)
 {
-    struct timespec ts;
+    struct timespec64 ts;
 
-    jiffies_to_timespec(jiffies, &ts);
+    jiffies_to_timespec64(jiffies, &ts);
 
     *nseconds = ((NvU64)ts.tv_sec * NSEC_PER_SEC + (NvU64)ts.tv_nsec);
 }
