diff -ruN a/drivers/hid/hid-input.c b/drivers/hid/hid-input.c
--- a/drivers/hid/hid-input.c	2021-04-03 10:29:42.158481203 +0200
+++ b/drivers/hid/hid-input.c	2021-04-03 10:37:39.350599528 +0200
@@ -1856,16 +1856,6 @@
 	list_for_each_entry(hidinput, &hid->inputs, list) {
 		if (hidinput->application == report->application)
 			return hidinput;
-
-		/*
-		 * Keep SystemControl and ConsumerControl applications together
-		 * with the main keyboard, if present.
-		 */
-		if ((report->application == HID_GD_SYSTEM_CONTROL ||
-		     report->application == HID_CP_CONSUMER_CONTROL) &&
-		    hidinput->application == HID_GD_KEYBOARD) {
-			return hidinput;
-		}
 	}
 
 	return NULL;
