diff -ruN a/Makefile b/Makefile
--- a/Makefile	2018-10-17 10:49:58.299992053 +0200
+++ b/Makefile	2018-10-17 10:50:37.159991983 +0200
@@ -802,7 +802,7 @@
 KBUILD_CFLAGS += -Wdeclaration-after-statement
 
 # Variable Length Arrays (VLAs) should not be used anywhere in the kernel
-KBUILD_CFLAGS += $(call cc-option,-Wvla)
+# KBUILD_CFLAGS += $(call cc-option,-Wvla)
 
 # disable pointer signed / unsigned warnings in gcc 4.0
 KBUILD_CFLAGS += -Wno-pointer-sign
