diff -ruN a/drivers/net/wireless/realtek/rtl88x2bu/os_dep/linux/os_intfs.c b/drivers/net/wireless/realtek/rtl88x2bu/os_dep/linux/os_intfs.c
--- a/drivers/net/wireless/realtek/rtl88x2bu/os_dep/linux/os_intfs.c	2019-03-21 20:39:47.377964776 +0100
+++ b/drivers/net/wireless/realtek/rtl88x2bu/os_dep/linux/os_intfs.c	2019-03-21 20:41:06.166761517 +0100
@@ -1289,14 +1289,10 @@
 	return dscp >> 5;
 }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0))
+	/* select_queue_fallback_t fallback ist rausgefallen
+	 */
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb,
-			    struct net_device *sb_dev,
-			    select_queue_fallback_t fallback)
-#else
-static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb,
-			    void *accel_priv, select_queue_fallback_t fallback)
-#endif
+			    struct net_device *sb_dev)
 {
 	_adapter	*padapter = rtw_netdev_priv(dev);
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
